healthcheck.php

```diff
@@ -0,0 +1,74 @@
+<?php
+namespace Automattic\VIP;
+
+/**
+ * Utility Healthcheck class that runs further than cache-healthcheck,
+ * and allows to verify the app code is working as expected.
+ */
+class Healthcheck {
+	/**
+	 * By default, an app is considered healthy unless filter value returns a non-empty array
+	 *
+	 * @var array
+	 */
+	protected array $errors = [];
+
+	public function __construct() {
+		$this->check();
+	}
+
+	/**
+	 * Apply the filters to and set the `$errors` property accordingly.
+	 *
+	 * @return void
+	 */
+	public function check() {
+		/**
+		 * Filters the errors that would indicate the app is unhealthy.
+		 *
+		 * @param string[]     $errors either empty array (no errors) or array of strings: [ 'an error has occurred', 'another error' ]
+		 */
+		$this->errors = apply_filters( 'vip_site_healthcheck_errors_array', $this->errors );
+		if ( ! is_array( $this->errors ) ) {
+			$this->errors = [ 'Unexpected value in Healthcheck->errors' ];
+		}
+	}
+
+	/**
+	 * Check if the app is healthy.
+	 * @return boolean true
+	 */
+	public function is_healthy() {
+		return is_array( $this->errors ) && empty( $this->errors );
```

Comment:
```suggestion
		return count( $this->errors ) === 0;
```

The array check seems redundant, as it is instantiated as an array, and there's a guard clause when assigning to it that if it isn't an array, then it's made an array.

See https://beberlei.de/2021/02/19/when_to_use_empty_in_php_i_say_never.html for why it's clearer code to stay away from `empty()`.